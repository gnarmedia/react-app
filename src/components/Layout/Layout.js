import React from "react";
import { Link, Switch, Route } from "react-router-dom";
import Home from "../Home/Home";
import About from "../About";
import Contact from "../Contact";

const Layout = () => (
  <div>
    <nav className="navbar" role="navigation" aria-label="main navigation">
      <div className="navbar-brand">
        <a className="navbar-item" href="https://bulma.io">
          <img
            src="https://bulma.io/images/bulma-logo.png"
            width="112"
            height="28"
          />
        </a>

        <a
          role="button"
          className="navbar-burger burger"
          aria-label="menu"
          aria-expanded="false"
          data-target="navbarBasicExample"
        >
          <span aria-hidden="true" />
          <span aria-hidden="true" />
          <span aria-hidden="true" />
        </a>
      </div>

      <div id="navbarBasicExample" className="navbar-menu">
        <div className="navbar-start">
          <Link to="/" className="navbar-item">
            Home
          </Link>

          <Link to="/about" className="navbar-item">
            About
          </Link>

          <Link to="/contact" className="navbar-item">
            Contact
          </Link>
        </div>

        <div className="navbar-end">
          <div className="navbar-item">
            <div className="buttons">
              <a className="button is-primary">
                <strong>Sign up</strong>
              </a>
              <a className="button is-light">Log in</a>
            </div>
          </div>
        </div>
      </div>
    </nav>

    <div className="container">
      <Switch>
        <Route path="/" exact component={Home} />
        <Route path="/about" exact component={About} />
        <Route path="/contact" exact component={Contact} />
      </Switch>
    </div>
  </div>
);

export default Layout;
