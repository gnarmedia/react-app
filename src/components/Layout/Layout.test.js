/**
 * @file Layout component tests
 */

import React from "react";
import { shallow } from "enzyme";

import Layout from "./Layout";

describe("Layout", () => {
  it("should render the Layout component", () => {
    const wrapper = shallow(<Layout />);

    expect(wrapper).toMatchSnapshot();
  });
});
