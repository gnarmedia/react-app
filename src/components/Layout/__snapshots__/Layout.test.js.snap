// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render the Layout component 1`] = `
<div>
  <nav
    aria-label="main navigation"
    className="navbar"
    role="navigation"
  >
    <div
      className="navbar-brand"
    >
      <a
        className="navbar-item"
        href="https://bulma.io"
      >
        <img
          height="28"
          src="https://bulma.io/images/bulma-logo.png"
          width="112"
        />
      </a>
      <a
        aria-expanded="false"
        aria-label="menu"
        className="navbar-burger burger"
        data-target="navbarBasicExample"
        role="button"
      >
        <span
          aria-hidden="true"
        />
        <span
          aria-hidden="true"
        />
        <span
          aria-hidden="true"
        />
      </a>
    </div>
    <div
      className="navbar-menu"
      id="navbarBasicExample"
    >
      <div
        className="navbar-start"
      >
        <Link
          className="navbar-item"
          replace={false}
          to="/"
        >
          Home
        </Link>
        <Link
          className="navbar-item"
          replace={false}
          to="/about"
        >
          About
        </Link>
        <Link
          className="navbar-item"
          replace={false}
          to="/contact"
        >
          Contact
        </Link>
      </div>
      <div
        className="navbar-end"
      >
        <div
          className="navbar-item"
        >
          <div
            className="buttons"
          >
            <Link
              className="button is-primary"
              replace={false}
              to="/sign-up"
            >
              <strong>
                Sign up
              </strong>
            </Link>
            <Link
              className="button is-light"
              replace={false}
              to="/sign-in"
            >
              <strong>
                Sign in
              </strong>
            </Link>
          </div>
        </div>
      </div>
    </div>
  </nav>
  <div
    className="container"
  >
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/about"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/contact"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/sign-in"
      />
      <Route
        component={[Function]}
      />
    </Switch>
  </div>
</div>
`;
